app/src/main/java/org/sagebionetworks/research/mpower/MpMainActivity.java

```diff
@@ -0,0 +1,85 @@
+package org.sagebionetworks.research.mpower;
+
+import android.annotation.SuppressLint;
+import android.content.Intent;
+import android.net.Uri;
+import android.os.Bundle;
+import android.support.v7.app.AppCompatActivity;
+import android.util.Log;
+import org.researchstack.backbone.factory.IntentFactory;
+import org.researchstack.backbone.task.Task;
+import org.sagebionetworks.research.researchStack.MpDataProvider;
+import org.sagebionetworks.research.researchStack.MpResourceManager;
+import org.sagebionetworks.research.researchStack.MpTaskFactory;
+import rx.Subscription;
+
+public class MpMainActivity extends AppCompatActivity {
+    String testPhoneNumber = "2063066209";
+    public static final int SIGN_UP_TASK_CODE = 1600;
+    String token = "252-345";
+
+    protected MpTaskFactory taskFactory = new MpTaskFactory();
+
+    @SuppressLint("RxLeakedSubscription")
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        handleIntent(getIntent());
+
+    }
+
+    @Override
+    protected void onResume() {
+        super.onResume();
+
+    }
+
+    protected void startSignupTask() {
+        Task signupTask = taskFactory.createTask(this, MpResourceManager.SIGNUP_TASK_RESOURCE);
+        Intent intent = IntentFactory.INSTANCE.newTaskIntent(
+                this, MpSignupActivity.class, signupTask);
+        startActivityForResult(intent, SIGN_UP_TASK_CODE);
+    }
+
+    protected void startOverviewActivity() {
+        // TODO define this
+    }
+
+    protected void startConsentActivity() {
+        // For now just launching to a link
+        Intent browserIntent = new Intent(Intent.ACTION_VIEW, Uri.parse("http://mpower.sagebridge.org/"));
```

Comment:
fyi they want the in-app consent to start at "http://mpower.sagebridge.org/study/intro".